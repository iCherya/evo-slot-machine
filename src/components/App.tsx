import React from 'react';

import { Layout } from '@/components/Layout/Layout';
import { Game } from '@/components/Game/Game';

export const App: React.FC = () => {
  return (
    <Layout>
      <Game />
    </Layout>
  );
};
