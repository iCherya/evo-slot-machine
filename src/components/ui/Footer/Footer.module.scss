@import '@/styles/variables.scss';

.wrapper {
  display: flex;
  justify-content: space-between;
  align-items: center;
  color: $color-black;
}

.socialButton {
  margin-left: $spacing-small;
}

.controlsButton {
  padding: 0;
  width: $font-size-xlarge;
  height: $font-size-xlarge;
  background-color: transparent;
  border: none;
  background-image: url('/icons/settings.svg');
  background-size: cover;
  background-repeat: no-repeat;
  transition: transform 0.3s ease-in-out;

  &:hover {
    transform: rotate(45deg);
    cursor: pointer;
  }
}
