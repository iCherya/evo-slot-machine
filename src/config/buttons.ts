export const BUTTON_OPTIONS = {
  SOCIAL: 'social',
  DEFAULT: 'default',
};
