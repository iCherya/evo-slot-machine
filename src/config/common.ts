export const COMMON = {
  buttonOptions: { SOCIAL: 'social', DEFAULT: 'default' },
  social: [
    { id: 0, type: 'github', link: 'https://github.com/icherya' },
    { id: 1, type: 'linkedin', link: 'https://www.linkedin.com/in/icherya/' },
  ],
};
