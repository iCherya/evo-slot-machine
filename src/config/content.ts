export const CONTENT = {
  header: {
    title: 'Jumbo Slot',
  },
};
