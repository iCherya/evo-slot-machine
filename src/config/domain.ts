export const DOMAIN = {
  currency: 'â‚¬',
  depositValue: 100,
  betValue: 10,
  rowWinMultiplier: 2,
  reelsCount: 3,
  turnoverCount: 10,
  turnoverDuration: 200,
  slotsConfig: [
    { id: 1, price: 4, image: 'assets/slots/slot-1-512.png', name: 'cherry', isWin: false },
    { id: 2, price: 8, image: 'assets/slots/slot-2-512.png', name: 'dollar', isWin: false },
    { id: 3, price: 16, image: 'assets/slots/slot-3-512.png', name: 'seven', isWin: false },
    { id: 4, price: 32, image: 'assets/slots/slot-4-512.png', name: 'grape', isWin: false },
    { id: 5, price: 10, image: 'assets/slots/slot-5-512.png', name: 'bar', isWin: false },
  ],
};
