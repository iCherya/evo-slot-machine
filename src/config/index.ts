export { DOMAIN } from './domain';
export { COMMON } from './common';
