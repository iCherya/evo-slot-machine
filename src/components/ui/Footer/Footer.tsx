import React, { useCallback } from 'react';

import { CONTENT } from '@/config/content';
import { BUTTON_OPTIONS } from '@/config/buttons';
import { Button } from '@/components/shared/Button';

import evolutionLogo from '/assets/evolution-logo.svg';

import styles from './Footer.module.scss';

export const Footer: React.FC = () => {
  const { description, social } = CONTENT.ui.footer;

  const onSettingsButtonClick = useCallback(() => {
    console.log('Settings button clicked');
  }, []);

  return (
    <footer className={styles.footer}>
      <div data-testid="credentials">
        <img src={evolutionLogo} alt="Evolution logo" />
        <p>{description}</p>
      </div>

      <button data-testid="settings-button" className={styles.controlsButton} onClick={onSettingsButtonClick} />

      <div data-testid="social-buttons">
        {social.map(({ id, type, link }) => (
          <span key={id} className={styles.socialButton}>
            <Button option={BUTTON_OPTIONS.SOCIAL} type={type} href={link} />
          </span>
        ))}
      </div>
    </footer>
  );
};
