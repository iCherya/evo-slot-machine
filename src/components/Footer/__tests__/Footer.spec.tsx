import { render } from '@testing-library/react';

import { Footer } from '@/components/Footer';
import { CONTENT } from '@/config/content';

jest.mock('@/components/shared/Button', () => ({
  Button: () => <div data-testid="button" />,
}));

describe('Footer', () => {
  it('should render credentials section', () => {
    const { getByTestId } = render(<Footer />);

    const credentials = getByTestId('credentials');

    expect(credentials.firstChild?.nodeName).toBe('IMG');
    expect(credentials.lastChild?.textContent).toBe(CONTENT.ui.footer.description);

    expect(credentials).toMatchSnapshot();
  });

  it('should render settings button', () => {
    const { getAllByTestId } = render(<Footer />);

    const settingsButton = getAllByTestId('settings-button');
    expect(settingsButton.length).toBe(1);

    expect(settingsButton).toMatchSnapshot();
  });

  it('should render social buttons', () => {
    const { getByTestId, getAllByTestId } = render(<Footer />);

    const socialButtons = getByTestId('social-buttons');
    expect(socialButtons.children.length).toBe(2);

    const socialButtonComponents = getAllByTestId('button');
    expect(socialButtonComponents.length).toBe(2);

    socialButtonComponents.forEach((socialButtonComponent, index) => {
      expect(socialButtonComponent.parentElement).toBe(socialButtons.children[index]);
    });

    expect(socialButtons).toMatchSnapshot();
  });
});
