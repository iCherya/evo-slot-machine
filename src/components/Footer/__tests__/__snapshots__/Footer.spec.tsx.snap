// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render credentials section 1`] = `
<div
  data-testid="credentials"
>
  <img
    alt="Evolution logo"
  />
  <p>
    TypeScript Bootcamp
  </p>
</div>
`;

exports[`Footer should render settings button 1`] = `
Array [
  <button
    data-testid="settings-button"
  />,
]
`;

exports[`Footer should render social buttons 1`] = `
<div
  data-testid="social-buttons"
>
  <span>
    <button
      data-testid="button"
    />
  </span>
  <span>
    <button
      data-testid="button"
    />
  </span>
</div>
`;
